server/src/main/java/org/apache/iotdb/db/engine/settle/SettleRequestHandler.java

```diff
@@ -0,0 +1,228 @@
+    return SettleRequestHandlerHolder.INSTANCE;
+  }
+
+  public TSStatus handleSettleRequest(TSettleReq req) {
+    List<String> paths = req.getPaths();
+
+    boolean hasSeqFile = false, hasUnSeqFile = false, hasModsFile = false;
+    Integer dataRegionId = null;
+    DataRegion dataRegion = null;
+    Integer level = null;
+    String storageGroupName = null;
+    Long timePartitionId = null;
+
+    Set<String> tsFileNames = new HashSet<>();
+
+    for (String path : paths) {
+      File tsFile = new File(path);
+      if (!tsFile.exists()) {
+        return RpcUtils.getStatus(TSStatusCode.PATH_NOT_EXIST, "file not exist.");
+      }
+      File modsFile = new File(path + ModificationFile.FILE_SUFFIX);
+      hasModsFile |= modsFile.exists();
+
+      int fileDataRegionId = TsFileUtils.getDataRegionId(tsFile);
+      if (dataRegion == null) {
+        dataRegionId = fileDataRegionId;
+        dataRegion = StorageEngine.getInstance().getDataRegion(new DataRegionId(dataRegionId));
+      } else if (dataRegionId != fileDataRegionId) {
+        return RpcUtils.getStatus(
+            TSStatusCode.ILLEGAL_PATH, "DataRegion of files is not consistent.");
+      }
+
+      String sgOfFile = TsFileUtils.getStorageGroup(tsFile);
+      if (storageGroupName == null) {
+        storageGroupName = sgOfFile;
+      } else if (!storageGroupName.equals(sgOfFile)) {
+        return RpcUtils.getStatus(
+            TSStatusCode.ILLEGAL_PATH, "StorageGroup of files is not consistent.");
+      }
+
+      long timePartitionOfFile = TsFileUtils.getTimePartition(tsFile);
+      if (timePartitionId == null) {
+        timePartitionId = timePartitionOfFile;
+      } else if (timePartitionId != timePartitionOfFile) {
+        return RpcUtils.getStatus(
+            TSStatusCode.ILLEGAL_PATH, "TimePartition of files is not consistent.");
+      }
+
+      String fileName = tsFile.getName();
+      TsFileNameGenerator.TsFileName tsFileName;
+      try {
+        tsFileName = TsFileNameGenerator.getTsFileName(fileName);
+      } catch (IOException e) {
+        return RpcUtils.getStatus(TSStatusCode.ILLEGAL_PATH, "Meet error when parsing TsFileName.");
+      }
+
+      int levelOfFile = tsFileName.getInnerCompactionCnt();
+      if (level == null) {
+        level = levelOfFile;
+      } else if (level != levelOfFile) {
+        return RpcUtils.getStatus(
+            TSStatusCode.ILLEGAL_PARAMETER, "Level of files is not consistent.");
+      }
+
+      if (TsFileUtils.isSequence(tsFile)) {
+        hasSeqFile = true;
+      } else {
+        hasUnSeqFile = true;
+      }
+      tsFileNames.add(fileName);
+      if (hasSeqFile && hasUnSeqFile) {
+        return RpcUtils.getStatus(
+            TSStatusCode.UNSUPPORTED_OPERATION, "Settle by cross compaction is not allowed.");
+      }
+    }
+
+    if (!hasModsFile) {
+      return RpcUtils.getStatus(
+          TSStatusCode.ILLEGAL_PARAMETER, "At least one mods file should be selected");
```

Comment:
Change the description because it may confuse the users. What we want to tell users is that every selected TsFile does not contains the mod file